src/init.cpp

```diff
@@ -1728,12 +1714,15 @@ bool AppInitMain(InitInterfaces& interfaces)
     if (gArgs.IsArgSet("-blocknotify"))
         uiInterface.NotifyBlockTip_connect(BlockNotifyCallback);
 
-    std::vector<fs::path> vImportFiles;
-    for (const std::string& strFile : gArgs.GetArgs("-loadblock")) {
-        vImportFiles.push_back(strFile);
-    }
+    if (gArgs.IsArgSet("-loadblock")) {
```

Comment:
Note that it is possible to run with `-noloadblock`. I suggest to add the condition before "spawning" the thread:
```cpp
if (!vImportFiles.empty()) {
    // Spawn 'loadblk' thread to run ThreadImport()
    threadGroup.create_thread(std::bind(&ThreadImport, vImportFiles));
}
```

Reply:
See https://github.com/bitcoin/bitcoin/blob/b2a6b0216192b6e8428f1a80b47f5178fccb961a/src/util/system.cpp#L407-L412

Basically you can invert options and so if you run with `-noloadblock` then `gArgs.IsArgSet("-loadblock") == true` but `gArgs.GetArgs("-loadblock")` gives an empty array. The result is that the import thread is spawned for no good reason.

Reply:
Thanks! Let me know if you see anything else that needs changing.